components/logins/ios/Logins/RustPasswordAPI.h

```diff
@@ -28,6 +28,11 @@ Sync15PasswordEngineRaw* sync15_passwords_state_new(char const* _Nonnull db_path
                                                     char const* _Nonnull encryption_key,
                                                     Sync15PasswordsError* _Nonnull error_out);
 
+Sync15PasswordEngineRaw* sync15_passwords_state_new_with_hex_key(char const* _Nonnull db_path,
```

Comment:
I vaguely recall you and @mhammond discussing names.  Please ensure whatever you agreed open makes it into here.  Personally, I would want this to just be `...with_key_bytes`; I think saying it's hex (and then interpreting upper and lower case hex differently, etc) is just asking for trouble later.  But I don't feel strongly about this.

Reply:
Right, the thing is, it's not the key bytes. E.g. providing the bytes of a string you used for sync15_password_state_new will fail, since the key is different.

We also don't have a choice about interpreting upper/lower hex differently. We're encoding those bytes as hex, and sending them to sqlcipher as a string. SQLcipher is the thing that treats upper/lower differently here.

The only way around this would be providing a raw key, but I'd rather not do that for the reasons here: https://github.com/mozilla/application-services/issues/255#issuecomment-449428767